src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestAttributeValueResolver.php

```diff
@@ -27,7 +27,13 @@
      */
     public function supports(Request $request, ArgumentMetadata $argument)
     {
-        return !$argument->isVariadic() && $request->attributes->has($argument->getName());
+        if ($argument->isVariadic()) {
+            return false;
+        }
+
+        $argumentName = $argument->getName();
+
+        return $request->attributes->has($argumentName) && null !== $request->attributes->get($argumentName);
```

Comment:
With this signature it would fail already, you just simply can't set it to `null` anymore and it will trigger the resolver.

```php
// call this with null, it will still trigger the resolver
public function fooAction(UserInterface $user = null);
```

But I'm still curious as of why the request values are null in the attribute bag.

@xabbuh regarding the above example, that would fail already. Also I don't think that values not mapped in the route should be affecting the parameter resolvers:
 - `/{required}` with `(string $required)` = allowed to be set to null but will throw an error as its required
 - `/` with `(string $required)` != allowed to set to null, as it was not added in the route definition